python_modules/dagster/dagster/core/definitions/event_metadata.py

```diff
@@ -330,19 +332,97 @@ def __new__(cls, asset_key: "AssetKey"):
         )
 
 
+@whitelist_for_serdes
+class TableSchemaMetadataEntryData(NamedTuple("_TableSchemaMetadataEntryData", [("schema", Dict)])):
+    """Representation of a schema for tabular data. Schema must be formatted as a
+    `Frictionless Table Schema<https://specs.frictionlessdata.io//table-schema/>`,
+    with the following modifications:
+
+    - The `type` of each field MAY be an arbitrary string (i.e. it is not restricted
+      to Frictionless types).
+    - Field `constraints` descriptor MAY contain a property `other` which
+      contains an array of strings. Each element should describe a constraint
+      that is not expressible with predefined Frictionless constraint types.
+    - A top-level property `constraints` MAY be included. This value is a
+      descriptor for "table-level" constraints. Presently only one property,
+      `other` is supported. This should contain a list of strings describing
+      arbitrary table-level constraints.
+
+    .. code-block:: python
+
+            # example schema
+            {
+                "constraints": {
+                    "other": [
+                        "foo > bar",
+                    ],
+                },
+                "fields": [
+                    {
+                        "name": "foo",
```

Comment:
I see that Frictionless distinguishes between "name" and "title", but I'm leery of including both. None of the in-memory or on-disk formats that I'm aware of have this distinction, so it feels like unnecessary complexity. I wouldn't argue that we should _never_ include it, but I'd rather wait until users ask for it before adding that surface area.